integrations/kube-agent-updater/pkg/testutils/mockserver.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testutils
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+type ServerMock struct {
+	Srv *httptest.Server
+
+	t        *testing.T
+	code     int
+	response string
+	path     string
+}
+
+func (m *ServerMock) SetResponse(t *testing.T, code int, response string) {
```

Comment:
can you set the response on the constructor to avoid data races?

Reply:
This would mean having to start a new server for each sub-test? This sounds a bit heavy compared to spinning one server and dynamically changing the response.

Reply:
It's just to protect if anyone runs the subtests tests in Parallel. You can also make the test parallel and add a comment saying the subtests must be sequential.